### DISCUSSION TITLE
Strongly Typed Ids and Collections

### DISCUSSION THREAD

**Discussion by Zumpel96** [riok/mapperly#1382] [General]
Hello.

I have a project with strongly typed ids. Everything works fine, if I want a simple mapping like this.

```csharp
public record StoreId
{
    public Guid Value { get; init; }
}

public record Store
{
    public StoreId Id { get; init; }

    public string Name { get; init; }
}

public record StoreDto
{
    public Guid Id { get; init; }

    public string Name { get; init; }
}

[Mapper]
public static partial class StoreDtoMapper
{
    [MapProperty($"{nameof(Store.Id)}.{nameof(StoreId.Value)}", nameof(StoreDto.Id))]
    public static partial StoreDto ToDto(this Store store);
}
```

However, I really struggle on mapping this with a collection now. I have another class, called product, which has a n:m relationship to stores (for simplicity only focus on product stores for now) and looks something like this.

```csharp
public record Product
{
    public ProductId Id { get; init; }

    public string Name { get; init; }

    public IReadonlyCollection<Store> Stores { get; init; }
}

public record ProductDto
{
    public Guid Id { get; init; }

    public string Name { get; init; }

    public List<StoreDto> Stores { get; init; }
}
```

The mapping works somewhat. The only problem I have is, that the Product.Stores[].Id.Value is not Mapped to ProductDto.Stores[].Id and returns a json like this.
```json
[
    {
        "id": "9d21ab4a-dbf3-4b37-94bd-b31a1fadaf5b",
        "name": "Testprodukt A",
        "stores": [
            {
                "id": "00000000-0000-0000-0000-000000000000",
                "name": "Test Store"
            }
        ]
    }
]
```

The id of the store is 100% not a Guid.Empty, so the mapping is just not considered.

I tried to play around with the "MapProperty" attribute, but didn't find the correct way on doing this. Is this a limitation of mapperly and I need to write a custom mapper? I hope, that someone can push me into the right direction.

**Comment by latonz** [comment]
Can you post the entire mapper definition, including the definitions for your product mappings? And the [generated code](https://mapperly.riok.app/docs/configuration/generated-source/)?

**Comment by latonz** [comment]
Closing as the requested information was not provided.
